# Following patch was applied to STM32U5's HAL OSPI driver to disable 
# optimizations. Compiling the app in -Os mode for certain STM32CubeIDE
# versions causes the driver to not function correctly.

diff --git a/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c b/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c
index 3d38b50..6ef32b0 100644
--- a/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c
+++ b/Drivers/STM32U5xx_HAL_Driver/Src/stm32u5xx_hal_ospi.c
@@ -1,3 +1,5 @@
+#pragma GCC push_options
+#pragma GCC optimize ("-O0")
 /**
   ******************************************************************************
   * @file    stm32u5xx_hal_ospi.c
@@ -3526,3 +3528,5 @@ HAL_StatusTypeDef HAL_OSPI_DLYB_GetClockPeriod(OSPI_HandleTypeDef *hospi, HAL_OS
   */
 
 #endif /* OCTOSPI || OCTOSPI1 || OCTOSPI2 */
+
+#pragma GCC pop_options
